### PR TITLE
Refactored alerts to be more clear

### PR DESCRIPTION
**Pull request by wrightmartin**
Fixes https://github.com/mysociety/alaveteli/issues/3291

- Using BEM syntax, so reusers can adapt, modify and build on them
- Using common terminology to help reusers and themers comprehend them better
- Build the country message entirely in JS (it relies on this to work anyway) to remove the needed to empty divs in the DOM and simplify styling

### CODE REVIEW

File: app/assets/javascripts/general.js [hunk 1]
```diff
@@ -1,31 +1,46 @@
 $(document).ready(function() {
  // flash message for people coming from other countries
- if(window.location.search.substring(1).search("country_name") == -1) {
-    if (!$.cookie('has_seen_country_message')) {
-  $.ajax({
-    url: "/country_message",
-        dataType: 'html',
-        success: function(country_message){
+ var htmlWrapperFront = [
```
Comment (wrightmartin) [inline app/assets/javascripts/general.js]: my JavaScript is *very* rusty. This is likely bad code, feel free to rip it apart

**Comment by garethrees** [comment]
Does this fix https://github.com/mysociety/alaveteli/pull/3346, or is there more that needs doing?

**Comment by wrightmartin** [comment]
@garethrees it fixes #3291, I believe. Unless there are notices and other messages I haven't been able to reproduce yet

**Comment by garethrees** [comment]
I think this needs a bit of a rethink. It looks like you were intending to change the [flash messages](https://github.com/mysociety/alaveteli/blob/develop/app/views/layouts/default.html.erb#L101-L106) (which I think is what we should call them, not alerts or popups) but you've actually changed the "[popup](https://github.com/mysociety/alaveteli/blob/develop/app/views/layouts/default.html.erb#L79-L87)". I actually like the name of that, as its an element that can be closed (which is what you associate with a popup).

You can set a popup by adding some content to [`app/views/general/_popup_banner.html.erb`](https://github.com/mysociety/alaveteli/blob/develop/app/views/general/_popup_banner.html.erb).

**Comment by wrightmartin** [comment]
Here's the notes from our chat about this

---
**Flash messages (rails terminology)**
- Undo the rename from flash to alerts
- Telling people something’s happened (error, success, warning, validation)
- Flash partials - boolean driven flash message that uses a partial to render

**Popups**
- Showing you're visiting the wrong country
- Survey CTA, Donate CTA
- Dismissible + cookie setting

**Form validation message**

- errorExplanation

**Comment by lizconlan** [comment]
rebased in hope of fixing travis fail

**Comment by lizconlan** [comment]
I think the `alaveteli` file was included in error

File: app/assets/javascripts/general.js (hunk 1 above)
Comment (lizconlan) [inline app/assets/javascripts/general.js]: ha ha - like mine's any better!

**Comment by lizconlan** [comment]
@wrightmartin code looks good to me, apart from the accidental file include (easily fixed). Can I assume that you've done the browserstack thing or should I? (either's fine!)

File: app/assets/javascripts/general.js [hunk 2]
```diff
@@ -1,31 +1,46 @@
...
-   $.cookie('seen_foi2', 1, { expires: 7, path: '/' });
-   return false;
-   });
+      }
+    })
+  }
+}
+
+ // popups
+$('#locality-popup .js-popup__close').click(function() {
```
Comment (lizconlan) [inline app/assets/javascripts/general.js]: This doesn't work as the block that creates `$('#locality-popup .js-popup__close')` is async (I think) and may still be running by the time you get here, meaning there's no object to handle the click event for. It fails silently, breaking the close button 😢 

Moving this inside the `success:` block, just after the closing `}` of `if (country_message)` seems to fix it.

File: alaveteli [hunk 3]
```diff
@@ -0,0 +1 @@
+/home/vagrant/alaveteli
```
Comment (lizconlan) [inline alaveteli]: this file included in error?

**Comment by lizconlan** [comment]
@wrightmartin I rebased it BTW, just so that the tests would pass - I recommend a git pull before you start work on it (sorry)

File: alaveteli (hunk 3 above)
Comment (wrightmartin) [inline alaveteli]: ah, yes - looks like something that shouldn't be here at all

**Comment by wrightmartin** [comment]
The weird styling is due to alavetelitheme's styles and alaveteli core's styles conflicting. I'll make sure this can't happen when other themes are using the updated core

**Comment by wrightmartin** [comment]
Thanks for the feedback liz, I've made all the changes you recommended

**Pull request merged**